src/main/java/org/mitre/synthea/engine/Generator.java

```diff
@@ -355,7 +356,89 @@ public Person generatePerson(int index, long personSeed) {
     }
     return person;
   }
+  
+  /**
+   * Generate a completely random Clinician. 
+   * The seed used to generate the person is randomized as well.
+   * 
+   * @param index Target index in the whole set of people to generate
+   * @return generated Person
+   */
+  public static Clinician generateClinician(int index, Provider provider) {
+    // System.currentTimeMillis is not unique enough
+    long clinicianSeed = UUID.randomUUID().getMostSignificantBits() & Long.MAX_VALUE;
+    return generateClinician(index, clinicianSeed, provider);
+  }
 
+  /**
+   * Generate a random clinician, from the given seed. 
+   * 
+   * @param index
+   *          Target index in the whole set of people to generate
+   * @param clinicianSeed
+   *          Seed for the random clinician
+   * @return generated Clinician
+   */
+  public static Clinician generateClinician(int index, long clinicianSeed, Provider provider) {
+    Clinician clinician = null;
+    try {
+      
+      // NOTE: Currently used default demographics for clinicians
+      
+      //Random randomForDemographics = new Random(clinicianSeed);
+      //Demographics city = location.randomCity(randomForDemographics);
+      Map<String, Object> out = new HashMap<>();
+
+      
+      //String race = city.pickRace(randomForDemographics);
```

Comment:
I think I understand why all these lines are commented out -- you want to use the same demographics for the city that the provider is in, and there's no easy way to do that at present, right? -- but could you add more comments explaining that so someone seeing this fresh would understand what's going on?